.travis.yml

```diff
@@ -1,22 +1,27 @@
 language: android
 
-android:
-  components:
-    - tools
-    - platform-tools
-    - build-tools-24.0.1
-    - android-24
-    - extra-android-m2repository
-    - sys-img-armeabi-v7a-android-15
-
 jdk:
   - oraclejdk8
 
+before_install:
+  # Install SDK license so Android Gradle plugin can install deps.
+  - mkdir "$ANDROID_HOME/licenses" || true
+  - echo "d56f5187479451eabf01fb78af6dfcb131a6481e" > "$ANDROID_HOME/licenses/android-sdk-license"
+  # Install the rest of tools (e.g., avdmanager)
+  - sdkmanager tools
+  # Install the system image
+  - sdkmanager "system-images;android-18;default;armeabi-v7a"
+  # Create and start emulator for the script. Meant to race the install task.
+  - echo no | avdmanager create avd --force -n test -k "system-images;android-18;default;armeabi-v7a"
+  - $ANDROID_HOME/emulator/emulator -avd test -no-audio -no-window &
+
+install: ./gradlew clean assemble assembleAndroidTest --stacktrace
+
 before_script:
-  - echo no | android create avd --force -n test -t android-15 --abi armeabi-v7a
-  - emulator -avd test -no-skin -no-audio -no-window &
   - android-wait-for-emulator
-  - adb shell input keyevent 82 &
+  - adb shell input keyevent 82
+
+script: ./gradlew check connectedCheck --stacktrace
```

Comment:
Travis setup was slightly weird since it was never running anything other than the `deploy_snapshot` script. Copied this file from Butterknife.

Reply:
It used the default config which runs `build` on Gradle projects.